pkgs/development/compilers/gcc/default.nix

```diff
@@ -74,6 +75,18 @@ let
   is10 = majorVersion == "10";
   is9  = majorVersion == "9";
 
+  # releases have a form: MAJOR.MINOR.MICRO, like 14.2.1
+  # snapshots have a form like MAJOR.MINOR.MICRO.DATE, like 14.2.1.20250322
+  isSnapshot = lib.length (lib.splitVersion version) == 4;
+  # return snapshot date of gcc's given version:
+  #   "14.2.1.20250322" -> "20250322"
+  #   "14.2.0" -> ""
+  snapDate = lib.concatStrings (lib.drop 3 (lib.splitVersion version));
+  # return base version without a snapshot:
+  #   "14.2.1.20250322" -> "14.2.0"
+  #   "14.2.0" -> "14.2.0"
+  baseVersion = lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
```

Comment:
Does upstream ever use the three‐dots convention here? Maybe `14.2.1-20250322` would be closer to the `14-20250322` naming of the tarballs and a little easier to parse?

Reply:
`gcc` reports the version via space character:

```
$ gcc --version | head -n1
gcc (GCC) 14.2.1 20250322
```

`a-b` is considered a single component for `lib.versions`:

```
nix-repl> lib.versions.pad 3 "1.2.3.4"
"1.2.3"

nix-repl> lib.versions.pad 3 "1.2.3-4"
"1.2.3-4"
```

Reply:
I expect that's a historical artifact from times when there were real three digits in the releases (before gcc-5*).